src/utils/backup.ts

```diff
@@ -0,0 +1,90 @@
+import { SoftwareKeyProvider } from 'jolocom-lib/js/vaultedKeyProvider/softwareProvider'
+// @ts-ignore
+import eccrypto from 'eccrypto'
+import { ICredentialAttrs } from 'jolocom-lib/js/credentials/credential/types'
+import { IKeyDerivationArgs } from 'jolocom-lib/js/vaultedKeyProvider/types'
+
+export interface BackupFile {
+  keys: EncryptedKey[]
+  data: string
+}
+
+export interface EncryptedKey {
+  pubKey: string
+  cipher: string
+}
+
+export async function createBackup(
+  did: string,
+  credentials: ICredentialAttrs[],
+  vault: SoftwareKeyProvider,
+  derivationArgs: IKeyDerivationArgs,
+): Promise<string> {
+  const data = {
```

Comment:
About this, can we make things more generic? basically take `data` directly as `any` input so that this can be reused in other contexts

Reply:
This would mean it becomes a "encryptHybrid" function or similar, right? it could also be moved to the key provider than